### PR TITLE
Add `BGRA8UNORM_STORAGE` extension

### PR DESCRIPTION
**Pull request by jinleili**
**Checklist**

- [x] Run `cargo clippy`.
- [x] Run `RUSTFLAGS=--cfg=web_sys_unstable_apis cargo clippy --target wasm32-unknown-unknown` if applicable.
- [ ] Add change to CHANGELOG.md. See simple instructions inside file.

**Connections**
resolves https://github.com/gfx-rs/wgpu/issues/3359

**Testing**
Tested locally on metal, vulkan(--features=vulkan-portability) and dx12 backends:
```rust
fn required_features() -> wgpu::Features {
    // wgpu::Features::TEXTURE_ADAPTER_SPECIFIC_FORMAT_FEATURES
    // OR
    wgpu::Features::BGRA8UNORM_STORAGE
}

// ...

let a_texture = device.create_texture(&wgpu::TextureDescriptor {
    format: wgpu::TextureFormat::Bgra8Unorm,
    usage: wgpu::TextureUsages::STORAGE_BINDING,
    ...,
});

let test_tv = a_texture.create_view(&wgpu::TextureViewDescriptor {
    format: Some(wgpu::TextureFormat::Bgra8Unorm),
    ..Default::default()
});
```

### CODE REVIEW

**Review by teoxoy — CHANGES_REQUESTED** [review]
Thanks for the PR!

I think we should wait for naga to add support for the texel format before we merge this so that we can test it.

File: wgpu-hal/src/vulkan/adapter.rs
Comment (jinleili) [reply wgpu-hal/src/vulkan/adapter.rs:529]: I'm sorry, I don't know how to do the check of `VK_FORMAT_FEATURE_2_STORAGE_WRITE_WITHOUT_FORMAT_BIT `. Check for `FORMAT_FEATURE_2` requires obtaining `FormatProperties3` first, but no relevant method was found in `ash`.

Comment (teoxoy) [reply wgpu-hal/src/vulkan/adapter.rs:529]: Isn't it this structure: https://docs.rs/ash/latest/ash/vk/struct.FormatProperties3.html ?

Comment (jinleili) [reply wgpu-hal/src/vulkan/adapter.rs:529]: Yes, but no way to get it is provided, only`get_physical_device_format_properties` and `get_physical_device_format_properties` to get `FormatProperties` `FormatProperties2`.

Comment (teoxoy) [reply wgpu-hal/src/vulkan/adapter.rs:529]: Hmm, how I think this should work is we call `get_physical_device_format_properties2` and pass it a `FormatProperties2` with a `FormatProperties3` in it's `pNext` chain.

This is quite convoluted as it requires extensions or specific vulkan versions.

`VK_KHR_get_physical_device_properties2` or VK1.1 for the query function and [`FormatProperties2`](https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/VkFormatProperties2.html) structure +
`VK_KHR_format_feature_flags2` or VK1.3 for [`FormatProperties3`](https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/VkFormatProperties3.html)

**Comment by cwfitzgerald** [comment]
Hey this got lost in the PR stack, could you rebase this?

With naga not supporting it yet, how about we land all the infrastructure, but we keep the feature disabled.

**Comment by jinleili** [comment]
@cwfitzgerald Already rebased, but I'm unsure how to keep the feature disabled.

**Comment by cwfitzgerald** [comment]
Probably through wgpu-core removing the feature after it queries the features from the device. Can remove it and link back to the naga issue.

**Comment by teoxoy** [comment]
There is still an unresolved discussion about the Vulkan impl.

**Review by cwfitzgerald — CHANGES_REQUESTED** [review]
Dismissing review request due to the above issue.

**Comment by nical** [comment]
Heads up, I'm continuing the work on this branch: https://github.com/nical/wgpu/tree/bgra8unorm-storage 
I'll propose change to this PR or open a new one soon (currently writing a test and hoping it'll work).

**Comment by jinleili** [comment]
I'm sorry I didn't deal with it for so long, since @nical is already done, it's more appropriate to open a new PR

**Pull request closed without merging**
